src/app/_components/NewMessageModal/NewMessageModal.tsx

```diff
@@ -0,0 +1,209 @@
+"use client";
+
+import {
+  Box,
+  Button,
+  IconButton,
+  Input,
+  InputAdornment,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import React, { useState, useEffect } from "react";
+import SearchIcon from "@mui/icons-material/Search";
+import CircularProgress from "@mui/material/CircularProgress";
+import CloseIcon from "@mui/icons-material/Close";
+
+interface User {
+  id: string;
+  name: string;
+  username: string;
+}
+
+interface NewMessageModalProps {
+  onSearch: (searchTerm: string) => Promise<User[]>;
+  onClose: () => void;
+  onNext: () => void;
+}
+
+const NewMessageModal: React.FC<NewMessageModalProps> = ({
+  onSearch,
+  onClose,
+  onNext,
+}) => {
+  const theme = useTheme();
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const [users, setUsers] = useState<User[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [searchInitiated, setSearchInitiated] = useState<boolean>(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const searchUsers = async () => {
+      if (!searchTerm) {
+        setUsers([]);
+        setSearchInitiated(false);
+        return;
+      }
+      setIsLoading(true);
+      try {
+        const results = await onSearch(searchTerm);
+        setUsers(results);
+        setSearchInitiated(true);
+      } catch (error) {
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    searchUsers().catch((error) =>
```

Comment:
you can probably put this catch inside the catch block instead of outside the try catch